### PR TITLE
Planetary atmos

### PR DESCRIPTION
**Pull request by MrsTonedOne**
:cl:
experiment: Atmos on planets is now more realistic. There is an unseen upper atmosphere that gas changes get carried away into, causing planet turfs to revert back to their original gas mixture.
tweak: This should massively improve atmos run speed as all of lavaland won't be processing the moment somebody opens a door anymore
/:cl:

Turfs marked as planetary_atmos will revert back to their default state as if the upper atmosphere mixes with it reverting it.

### CODE REVIEW

**Comment by lzimann** [comment]
no more 1000% atmos ss then?

**Comment by MrStonedOne** [comment]
exactly

**Comment by KorPhaeron** [comment]
Turns lavaland into space currently

**Comment by FlavoredCactus** [comment]
Bless you.

**Comment by MrStonedOne** [comment]
>Turns lavaland into space currently

Ya, I think I got that figured out.

**Comment by MrStonedOne** [comment]
seems to be good. no errors found on sybil

File: code/modules/mining/mine_turfs.dm [hunk 1]
```diff
@@ -619,10 +619,12 @@
 
 /turf/open/floor/plating/asteroid/basalt/lava_land_surface
 	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	planetary_atmos = TRUE
 	baseturf = /turf/open/floor/plating/lava/smooth/lava_land_surface
 
 /turf/open/chasm/straight_down/lava_land_surface
 	initial_gas_mix = "o2=14;n2=23;TEMP=300"
+	planetary_atmos = TRUE
```
Comment (coiax) [inline code/modules/mining/mine_turfs.dm]: Given that /turf/open/chasm/straight_down/lava_land_surface/normal_air is only used on the Snappop shuttle at the moment, it'd be weird for it to have planetary_atmos on a ship. How does planetary atmos interact with air leaking to space?

**Comment by Jalleo** [comment]
Wait does this mean mining could now start mining the air for atmos?

**Comment by coiax** [comment]
I'm not sure if Atmos or Science are particularly short on supplies of Nitrogen or Oxygen, but you are right, this would allow the station to actually harvest the air from the planet below. Probably be a lot of cannisters moving around. Not sure if Atmos pipes interact nicely with shuttles. but it would be a fun experiment.

**Comment by lzimann** [comment]
Pipes should work perfectly fine on shuttles, the only problem is that the smoothing used to break when shuttle moved.

**Comment by Dorsisdwarf** [comment]
If we get this going we can make some gases sellable in cargo, let cargo mine the very air of lavaland itself

File: code/modules/mining/mine_turfs.dm (hunk 1 above)
Comment (MrStonedOne) [inline code/modules/mining/mine_turfs.dm]: for the most part, i assumed the lava_land_surface suffix is only going to be used on the surface of the planet of lavaland.

`/turf/open/chasm/straight_down/lava_land_surface/normal_air` sounds like it should really be `/turf/open/chasm/straight_down/normal_air` what does the lava_land_surface child type give?

**Comment by FlavoredCactus** [comment]
Runtime after blowing up a bomb in the mining station on lavaland.

`[16:39:55]Cannot execute null.archive().
[16:39:55]proc name: archive (/turf/open/archive)
[16:39:55]  source file: LINDA_turf_tile.dm,92
[16:39:55]  usr: null
[16:39:55]  src: the chasm (54,30,5) (/turf/open/chasm/straight_down/lava_land_surface)
[16:39:55]  call stack:
[16:39:55]the chasm (54,30,5) (/turf/open/chasm/straight_down/lava_land_surface): archive()
[16:39:55]the chasm (53,30,5) (/turf/open/chasm/straight_down/lava_land_surface): process cell(90)
[16:39:55]Air (/datum/subsystem/air): process active turfs(0)
[16:39:55]Air (/datum/subsystem/air): fire(0)
[16:39:55]Master (/datum/controller/master): RunQueue()
[16:39:55]Master (/datum/controller/master): Loop()
[16:39:55]Master (/datum/controller/master): StartProcessing()`

**Comment by coiax** [comment]
@Dorsisdwarf I think we might have to ensure the gasses we're selling in cargo are in containers; no shipping a burning cargo shuttle going "LOOK, HIGH QUALITY PLASMA"

**Comment by MrStonedOne** [comment]
very likely to be unrelated FlavoredCactus.

It seems the turf's air is getting set to null, but this PR never assigns anything to air. I'll have to look at the logic tree for turfs getting deleted/created to figure out why an open turf is having no air datum. another pr idea i had was to make airs only exist for turfs with differences from their initial_atmos strings, this would fix the issue as that wouldn't be out of the normal, but that will have to wait for the overlays refactor to be finished.

**Pull request merged**